File: insertion_sort/insertion_sort1.py
def insertion_sort(A):
    for i in range(1, len(A)):
        pos = i
        for j in range(i, -1, -1):
            if(A[j]>A[i]):
                pos=j
        temp=A[i]
        for k in range(i, pos, -1):
            A[k]=A[k-1]
        A[pos]=temp
    return A


print(insertion_sort([1, 2, 10, 8, 9, 12, 11, 10, 28, 7]))

File: insertion_sort/InsertionSort1.java
class InsertionSort1{
    public static int[] insertionSort1(int[] A){
        for(int i=1;i<A.length;i++){
            int pos=i;
            for(int j=i;j>-1;j--){
                if(A[j]>A[i]){
                    pos=j;
                }
            }
            int temp=A[i];
            for(int j=i;j>pos;j--){
                A[j]=A[j-1];
            }
            A[pos]=temp;
        }
        return A;
    }
    public static void main(String[] args){
        int[] A={
            1, 8, 6, 3, 10, 9
        };
        A=insertionSort1(A);
        for(int i=0;i<A.length;i++){
            System.out.print(A[i]+"\t");
        }
    }
}